src/packages/authentication/AuthServiceImpl.ts

```diff
@@ -10,49 +9,53 @@ import {
     createLogger
 } from "@open-pioneer/core";
 import type {
-    AuthEvents,
     AuthPlugin,
     AuthService,
     AuthState,
     LoginBehavior,
     SessionInfo
 } from "./api";
 import type { Service, ServiceOptions } from "@open-pioneer/runtime";
+import { syncWatch } from "@conterra/reactivity-core";
 
 const LOG = createLogger("authentication:AuthService");
 
-export class AuthServiceImpl extends EventEmitter<AuthEvents> implements AuthService, Service {
+export class AuthServiceImpl implements AuthService, Service {
     #plugin: AuthPlugin;
-    #currentState: AuthState;
     #whenUserInfo: ManualPromise<SessionInfo | undefined> | undefined;
-    #eventHandle: Resource | undefined;
+    #watchPluginStateHandle: Resource | undefined;
 
     constructor(serviceOptions: ServiceOptions<{ plugin: AuthPlugin }>) {
-        super();
         this.#plugin = serviceOptions.references.plugin;
 
-        // Init from plugin state and watch for changes.
-        this.#currentState = this.#plugin.getAuthState();
-        this.#eventHandle = this.#plugin.on?.("changed", () => this.#onPluginStateChanged());
+        this.#watchPluginStateHandle = syncWatch(
+            () => [this.#plugin.getAuthState()],
+            ([state]) => {
+                this.#onPluginStateChanged(state);
+            },
+            {
+                immediate: true
```

Comment:
Immediate is not necessary here. Removing it would restore the exact same behavior as before.